src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -818,21 +823,47 @@ void AtlasEngine::_recreateSizeDependentResources()
         _api.glyphProps = Buffer<DWRITE_SHAPING_GLYPH_PROPERTIES>{ projectedGlyphSize };
         _api.glyphAdvances = Buffer<f32>{ projectedGlyphSize };
         _api.glyphOffsets = Buffer<DWRITE_GLYPH_OFFSET>{ projectedGlyphSize };
-
-        D3D11_BUFFER_DESC desc;
-        desc.ByteWidth = gsl::narrow<u32>(totalCellCount * sizeof(Cell)); // totalCellCount can theoretically be UINT32_MAX!
-        desc.Usage = D3D11_USAGE_DYNAMIC;
-        desc.BindFlags = D3D11_BIND_SHADER_RESOURCE;
-        desc.CPUAccessFlags = D3D11_CPU_ACCESS_WRITE;
-        desc.MiscFlags = D3D11_RESOURCE_MISC_BUFFER_STRUCTURED;
-        desc.StructureByteStride = sizeof(Cell);
-        THROW_IF_FAILED(_r.device->CreateBuffer(&desc, nullptr, _r.cellBuffer.put()));
-        THROW_IF_FAILED(_r.device->CreateShaderResourceView(_r.cellBuffer.get(), nullptr, _r.cellView.put()));
     }
 
-    // We have called _r.deviceContext->ClearState() in the beginning and lost all D3D state.
-    // This forces us to set up everything up from scratch again.
-    _setShaderResources();
+    if (!_r.d2dMode)
+    {
+        // The RenderTargetView is later used with OMSetRenderTargets
+        // to tell D3D where stuff is supposed to be rendered at.
+        {
+            wil::com_ptr<ID3D11Texture2D> buffer;
+            THROW_IF_FAILED(_r.swapChain->GetBuffer(0, __uuidof(ID3D11Texture2D), buffer.put_void()));
+            THROW_IF_FAILED(_r.device->CreateRenderTargetView(buffer.get(), nullptr, _r.renderTargetView.put()));
+        }
+
+        // Tell D3D which parts of the render target will be visible.
+        // Everything outside of the viewport will be black.
+        //
+        // In the future this should cover the entire _api.sizeInPixel.x/_api.sizeInPixel.y.
+        // The pixel shader should draw the remaining content in the configured background color.

```

Comment:
i know this is not a new comment, but should we promote any of these unmarked TODOs to issues?

Reply:
Good point, because this comment is certainly outdated. We now support margins for the renderer (see the `backgroundColor` variable in `shader_ps.hlsl`).